### ISSUE TITLE
descriptors never found (noble adapter)

### ISSUE THREAD

**Issue by OJFord** [thegecko/bleat#3]
For example, modifying [the example](https://github.com/thegecko/bleat/blob/master/example_node.js#L17-L20) slightly:

```javascript
Object.keys(device.services).forEach(function(serviceID) {
    var service = device.services[serviceID];
    logStatus("found service: " + service.uuid + " (with " + Object.keys(service.characteristics).length + " characteristics)"));

    Object.keys(service.characteristics).forEach(function(charID) {
        var characteristic = service.characteristics[charID];
        logStatus("found characteristic:" + characteristic.uuid + " (with " + Object.keys(characteristic.descriptors).length + " descriptors)");
    });
});
```

Which logs "(with 0 descriptors)" always.

It seems the [discoverDescriptors callback](https://github.com/thegecko/bleat/blob/master/dist/bleat.noble.js#L114-L121) is "never" called.

May be related to sandeepmistry/noble#136, though the example above does not enable notifications first (or ever).

**Comment by thegecko** [comment]
This looks like a problem with the yosemite XPC communication in noble (or lower).

I'll dig some more and raise an issue.

**Comment by OJFord** [comment]
Sounds likely -- tried on a second machine, this one running El Capitan (beta), and it doesn't even get as far as that hurdle. It initialises okay, but the "found" callback of `startScan` never fires.

**Comment by thegecko** [comment]
I've run tests with the noble examples and descriptors are found!
I think it's a problem reading descriptors after using the noble ```discoverAllServicesAndCharacteristics``` function, so I'll rewrite the adapter to not use it.

**Comment by OJFord** [comment]
Ah yes sorry should have said it finds them okay in my non-bleat branch. Which supports that theory because for whatever reason, I used `discoverServices` and `discoverCharacteristics` separately. Cheers!

**Comment by thegecko** [comment]
Should now be resolved in #0.0.13
